src/librustc/lint/builtin.rs

```diff
@@ -224,6 +224,12 @@ declare_lint! {
 }
 
 declare_lint! {
+    pub LEGACY_CONSTRUCTOR_VISIBILITY,
+    Deny,
```

Comment:
Do we need to make this deny-by-default, given that there are three breaking crates in the wild?
While the warning cycle is a bit hacky, I think the risk of "false positives" is very low.